modin/experimental/core/execution/native/implementations/omnisci_on_native/exchange/dataframe_protocol/dataframe.py

```diff
@@ -0,0 +1,332 @@
+            yield col
+
+    def get_column(self, i: int) -> OmnisciProtocolColumn:
+        return OmnisciProtocolColumn(
+            OmnisciProtocolDataframe(
+                self._df.mask(col_positions=[i]),
+                allow_copy=self._allow_copy,
+            ),
+        )
+
+    def get_column_by_name(self, name: str) -> OmnisciProtocolColumn:
+        return OmnisciProtocolColumn(
+            OmnisciProtocolDataframe(
+                self._df.mask(col_labels=[name]),
+                allow_copy=self._allow_copy,
+            ),
+        )
+
+    def get_columns(self) -> Iterable[OmnisciProtocolColumn]:
+        for name in self._df.columns:
+            yield OmnisciProtocolColumn(
+                OmnisciProtocolDataframe(
+                    self._df.mask(col_labels=[name]),
+                    nan_as_null=self._nan_as_null,
+                    allow_copy=self._allow_copy,
+                ),
+            )
+
+    def select_columns(self, indices: Sequence[int]) -> "OmnisciProtocolDataframe":
+        if not isinstance(indices, collections.Sequence):
+            raise ValueError("`indices` is not a sequence")
+
+        return OmnisciProtocolDataframe(
+            self._df.mask(col_positions=list(indices)),
+            nan_as_null=self._nan_as_null,
+            allow_copy=self._allow_copy,
+        )
+
+    def select_columns_by_name(
+        self, names: Sequence[str]
+    ) -> "OmnisciProtocolDataframe":
+        if not isinstance(names, collections.Sequence):
+            raise ValueError("`names` is not a sequence")
+
+        return OmnisciProtocolDataframe(
+            self._df.mask(col_labels=list(names)),
+            nan_as_null=self._nan_as_null,
+            allow_copy=self._allow_copy,
+        )
+
+    def get_chunks(
+        self, n_chunks: Optional[int] = None
+    ) -> Iterable["OmnisciProtocolDataframe"]:
+        if n_chunks is None:
+            return self._yield_chunks(self._chunk_slices)
+
+        if n_chunks % self.num_chunks() != 0:
+            raise RuntimeError(
+                "The passed `n_chunks` has to be a multiple of `num_chunks`."
+            )
+
+        if n_chunks > self.num_rows():
+            raise RuntimeError(
+                "The passed `n_chunks` value is bigger than the amout of rows in the frame."
+            )
+
+        extra_chunks = n_chunks - self.num_chunks()
+        # `._chunk_slices` is a cached property, we don't want to modify the property's
+        # array inplace, so doing a copy here
+        subdivided_slices = self._chunk_slices.copy()
+
+        # The subdividing behavior is a bit different from "subdividing each chunk",
+        # instead it subdivides the biggest chunks first, so overall chunking be as
+        # equal as possible
+        for _ in range(extra_chunks):
+            # 1. Find the biggest chunk
+            # 2. Split it in the middle
+            biggest_chunk_idx = np.argmax(np.diff(subdivided_slices))
```

Comment:
Rewritten it into a "subdivide each chunk" maner. However, it may still be required to do this `O(N**2)` logic (for a significantly less N though) at the end in case of originally bad chunking. (see the updated `.get_chunks()` function for details)